pkg/asset/manifests/aws/zones_test.go

```diff
@@ -224,78 +256,163 @@ func Test_setSubnetsManagedVPC(t *testing.T) {
+					},
+					ZonesInRegion: []string{"a", "b", "c"},
+				},
+			},
+			want: capa.NetworkSpec{
+				VPC: capa.VPCSpec{CidrBlock: stubDefaultCIDR},
+				Subnets: []capa.SubnetSpec{
+					{
+						ID:               "infra-id-subnet-private-a",
+						AvailabilityZone: "a",
+						IsPublic:         false,
+						CidrBlock:        "10.0.0.0/19",
+					}, {
+						ID:               "infra-id-subnet-private-b",
+						AvailabilityZone: "b",
+						IsPublic:         false,
+						CidrBlock:        "10.0.32.0/19",
+					}, {
+						ID:               "infra-id-subnet-private-c",
+						AvailabilityZone: "c",
+						IsPublic:         false,
+						CidrBlock:        "10.0.64.0/19",
+					}, {
+						ID:               "infra-id-subnet-private-edge-a",
+						AvailabilityZone: "edge-a",
+						IsPublic:         false,
+						CidrBlock:        "10.0.128.0/21",
+					}, {
+						ID:               "infra-id-subnet-public-a",
+						AvailabilityZone: "a",
+						IsPublic:         true,
+						CidrBlock:        "10.0.96.0/21",
+					}, {
+						ID:               "infra-id-subnet-public-b",
+						AvailabilityZone: "b",
+						IsPublic:         true,
+						CidrBlock:        "10.0.104.0/21",
+					}, {
+						ID:               "infra-id-subnet-public-c",
+						AvailabilityZone: "c",
+						IsPublic:         true,
+						CidrBlock:        "10.0.112.0/21",
+					}, {
+						ID:               "infra-id-subnet-public-edge-a",
+						AvailabilityZone: "edge-a",
+						IsPublic:         true,
+						CidrBlock:        "10.0.136.0/21",
+					},
+				},
+			},
 		},
 	}
 	for _, tt := range tests {
-		tt := tt // TODO: remove with golang 1.22
 		t.Run(tt.name, func(t *testing.T) {
 			err := setSubnetsManagedVPC(tt.args.in)
 			if (err != nil) != tt.wantErr {
 				t.Errorf("setSubnetsManagedVPC() #1 error: %+v,\nwantErr %+v\n", err, tt.wantErr)
 			}
-			var got *capa.NetworkSpec
-			if tt.args.in != nil && tt.args.in.Cluster != nil {
-				got = &tt.args.in.Cluster.Spec.NetworkSpec
-			} else {
-				if !tt.wantErr {
-					t.Errorf("setSubnetsManagedVPC() #2 error: %v, wantErr: %v", err, tt.wantErr)
-				}
+			// TODO review those empty test cases
+			if tt.args.in == nil && (err == nil) {
+				return
+			}
+			if tt.args.in.Cluster == nil && (err == nil) {
 				return
 			}
-			if len(got.Subnets) == 0 {
+
+			if len(tt.args.in.Cluster.Spec.NetworkSpec.Subnets) == 0 {
 				if !tt.wantErr {
 					t.Errorf("setSubnetsManagedVPC() #2 error: %v, wantErr: %v", err, tt.wantErr)
 				}
 				return
 			}
-			got.Subnets = tSortCapaSubnetsByID(got.Subnets)
-			if tt.want != nil {
-				assert.EqualValuesf(t, tt.want, got, "%v failed.\nWant: %+v\nGot: %+v", tt.name)
+			tt.args.in.Cluster.Spec.NetworkSpec.Subnets = tSortCapaSubnetsByID(tt.args.in.Cluster.Spec.NetworkSpec.Subnets)
+			if got := tt.args.in.Cluster.Spec.NetworkSpec; !cmp.Equal(got, tt.want) {
+				t.Errorf("setSubnetsManagedVPC() NetworkSpec.Subnets diff: %v", cmp.Diff(got, tt.want))
```

Comment:
@r4f4 is there a better alternative available on installer of `cmp` providing a friendly `Diff` of complex structures?

Reply:
nvm, managed it.